pkg/lib/infra/sms/custom/client.go

```diff
@@ -94,6 +94,10 @@ func (w *SMSWebHook) Call(ctx context.Context, u *url.URL, payload SendOptions)
 
 	responseBody, err := ParseResponseBody(bodyBytes)
 	if err != nil {
+		// This is not something we understand, check the status code to determine if it is a success
+		if resp.StatusCode >= 200 && resp.StatusCode < 300 {
+			return nil
+		}
```

Comment:
We should check if the error is produced by the `json` package. Please add tests to ensure this does not break legacy gateways.

Reply:
Originally we don't even try to read the response and only check the statusCode. If we want to ensure it behave the same as before we should still treat it as success even it is not a json parsing error.